Sources/Datadog/DatadogCore/Upload/DataUploadWorker.swift

```diff
@@ -62,32 +62,38 @@ internal class DataUploadWorker: DataUploadWorkerType {
             if let batch = nextBatch {
                 DD.logger.debug("⏳ (\(self.featureName)) Uploading batch...")
 
-                // Upload batch
-                let uploadStatus = self.dataUploader.upload(
-                    events: batch.events,
-                    context: context
-                )
-
-                // Delete or keep batch depending on the upload status
-                if uploadStatus.needsRetry {
-                    self.delay.increase()
-
-                    DD.logger.debug("   → (\(self.featureName)) not delivered, will be retransmitted: \(uploadStatus.userDebugDescription)")
-                } else {
+                do {
+                    // Upload batch
+                    let uploadStatus = try self.dataUploader.upload(
+                        events: batch.events,
+                        context: context
+                    )
```

Comment:
All this diff is because I had to make `self.dataUploader.upload()` a `throwing` method..

Reply:
Good move 👍